doc/arch/adr-010-learn-to-rank.md

```diff
@@ -0,0 +1,286 @@
+# Decision record: Learning To Rank
+
+**Date:** 2019-11-07
+
+The search team have decided to implement [Learning To Rank][].
+
+1. What is Learning To Rank?
+1. Motivation
+1. Implementation
+1. What success will look like
+
+[Learning To Rank]: https://en.wikipedia.org/wiki/Learning_to_rank
+
+## What is Learning To Rank?
+
+Learning To Rank (LTR) is the application of machine learning to create a
+ranking model that can be used to improve search relevance.
+
+This is a common approach for advanced search applications.
+
+The ranking of search results on websites are frequently determined by
+machine learning models. For example, Airbnb, Yelp, Wikipedia, Bloomberg,
+and commercial search engines.
+
+There is a good explanation on what LTR is here:
+
+- https://elasticsearch-learning-to-rank.readthedocs.io/en/latest/core-concepts.html
+- https://opensourceconnections.com/blog/2017/02/24/what-is-learning-to-rank/
+
+### How does Learning To Rank work?
+
+It's easiest to explain with an example:
+
+Say you searched for 'Harry Potter'. Search API might return the following results:
+
+1. HMRC Sign in
+2. Harry Potter World
+3. Pottery barn
+4. Harry Potter
+
+These are not good results.
+
+Our approach now (prior to implementing Learning To Rank) would be to
+manually rank these results, giving them a score of between 0-3 with
+0 being irrelevant and 3 being completely relevant:
```

Comment:
This sounds like we determine the order of search results by manually ranking them.  Say something like

> To assess search quality, we can assign *relevancy judgements* to the results of a query in the range 0-3, with 0 being irrelevant and 3 being perfect.  For example, for these results: